### PR TITLE
Add Feedback pages for both view mode and submit mode

### PR DESCRIPTION
**Pull request by Spartan97**
This PR creates two pages linked to a puzzle. 

The first is a Feedback page that shows a table of previously submitted feedback items. It is meant to be the admin/author view to aggregate all the feedback received. Eventually this will also aggregate fun/difficulty ratings to give an overall puzzle score.

The second is a SubmitFeedback page that allows a user to submit a new feedback item. This page allows the user to give written feedback and rate fun and difficulty from 0-5. It also automatically pulls the current time. There is still work to do around pulling the user and team (once auth is complete) and allowing a user to edit feedback already submitted instead of creating multiples.

### CODE REVIEW

File: ServerCore/Pages/Puzzles/Feedback.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
```
Comment (morganbr) [inline ServerCore/Pages/Puzzles/Feedback.cshtml.cs]: Please run remove/sort usings

File: ServerCore/Pages/Puzzles/SubmitFeedback.cshtml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@model ServerCore.Pages.Puzzles.SubmitFeedbackModel
+
+@{
+    ViewData["Title"] = "Submit Feedback";
+    var id = ViewContext.RouteData.Values["id"];
+}
+
+<h2>Submit Feedback for @Html.DisplayFor(model => model.Puzzle.Name)</h2> 
+<!-- 
+    This part isn't quite working yet...
```
[the remark this answers is not in the document]
Comment (Spartan97) [reply ServerCore/Pages/Puzzles/SubmitFeedback.cshtml]: Whoops I got that working and forgot to remove my comment. Carry on!

File: ServerCore/Pages/Puzzles/Feedback.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            _context = context;
+        }
+
+        public IList<Feedback> Feedbacks { get; set; }
+        public Puzzle Puzzle { get; set; }
+
+        public async Task OnGetAsync(int id)
+        {
+            Feedbacks = await _context.Feedback.Where((f) => f.Puzzle.ID == id).ToListAsync();
+            Puzzle = await _context.Puzzles.Where((p) => p.ID == id).FirstOrDefaultAsync();
```
[the remark this answers is not in the document]
Comment (Spartan97) [reply ServerCore/Pages/Puzzles/Feedback.cshtml.cs]: Fixed in iteration 2.

File: ServerCore/Pages/Puzzles/Feedback.cshtml.cs (hunk 1 above)
Comment (Spartan97) [reply ServerCore/Pages/Puzzles/Feedback.cshtml.cs]: In correct order by accident -- System first then everything else alphabetized.

File: ServerCore/Pages/Puzzles/Feedback.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Puzzles
+{
+    public class FeedbackModel : EventSpecificPageModel
+    {
+        private readonly ServerCore.Models.PuzzleServerContext _context;
+
+        public FeedbackModel(ServerCore.Models.PuzzleServerContext context)
```
[the remark this answers is not in the document]
Comment (Spartan97) [reply ServerCore/Pages/Puzzles/Feedback.cshtml.cs]: I didn't see a lot elsewhere, and some of them were simple constructors, but I added a few to the methods that had useful code in them.

**Comment by Spartan97** [comment]
I believe I've fully addressed the comments from iteration one now!

**Comment by Spartan97** [comment]
@morganbr @tabascq Is there anything left to do here? I can't make tonight but would like to get this merged so I can submit a new PR in the next few days.

File: ServerCore/Pages/Puzzles/Feedback.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace ServerCore.Pages.Puzzles
+{
+    /// <summary>
+    /// Model for the author/admin's view of the feedback items for a specific puzzle
+    /// /used for viewing and aggregating feedback for a specific puzzle
+    /// </summary>
+    public class FeedbackModel : EventSpecificPageModel
+    {
+        private readonly ServerCore.Models.PuzzleServerContext _context;
```
Comment (morganbr) [inline ServerCore/Pages/Puzzles/Feedback.cshtml.cs]: PuzzleServerContext just moved to the ServerCore.DataModel namespace, so you'll need to remove namespace qualification where you're using it

File: ServerCore/Pages/Puzzles/SubmitFeedback.cshtml.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        /// Takes the filled out items and adds it to the database as a new piece of feedback.
+        /// </summary>
+        public async Task<IActionResult> OnPostAsync(int id)
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            Feedback.Puzzle = await _context.Puzzles.Where(m => m.ID == id).FirstOrDefaultAsync();
```
Comment (morganbr) [inline ServerCore/Pages/Puzzles/SubmitFeedback.cshtml.cs]: This could also use a validity check on the puzzle

**Comment by Spartan97** [comment]
Resolved merge conflicts!

**Pull request merged**